core/tests/puppeteer-acceptance-tests/user-utilities/super-admin-utils.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Super Admin users utility file.
+ */
+
+const puppeteerUtilities = require(
+  '../puppeteer-testing-utilities/puppeteer-utils.js');
+const testConstants = require(
+  '../puppeteer-testing-utilities/test-constants.js');
+const { showMessage } = require(
+  '../puppeteer-testing-utilities/show-message-utils.js');
+
+const rolesEditorTab = testConstants.URLs.RolesEditorTab;
+const roleEditorInputField = 'input.e2e-test-username-for-role-editor';
+const roleEditorButtonSelector = 'button.e2e-test-role-edit-button';
+const rolesSelectDropdown = 'div.mat-select-trigger';
+const addRoleButton = 'button.oppia-add-role-button';
+
+module.exports = class superAdmin extends puppeteerUtilities {
```

Comment:
It seems to a bit weird to extend utilities? Why do we do that and should we try to explain it more?

Reply:
I agree to your point @vojtechjelinek . These classes are user specific utiliies which extends the base puppeteerutilities. For eg. superadmin utilities as in this file adds methods specific to superadmin ( such as assigning roles, revoking roles etc. ) . This helps to keep code segregated on user basis. We can rename it to superAdminUtils but the same thing needs to be done to other files in the user-utils folder which also come under this point you mentioned.

Reply:
or maybe rename `puppeteerUtilities` so that it is clear that it can work as user base class

Reply:
can we rename the base class ( puppeteerUtilities ) to `browser` ?

Reply:
If you have an "extends" relationship, this implies an "is a" relationship. Would it be correct to say that a superAdmin "is a" browser?

Reply:
@seanlip @vojtechjelinek I think it all depends on the context. I would prefer to keep the name `puppeteerUtils` but this class no just acts as a utils class; instead it forms the base of all the tests. All the instances are generated through the base class `puppeteerUtils`, so in a sense it's a base class + utils class . But it would wise to call it base class overall, because the utilities present in those class are basic and they can also be segregated if needed. I think we need to reconsider our naming convention, because it's still pointing out some faults here and there. /cc @Shivkant-Chauhan

Reply:
Then should we call it something like `baseUser`? It makes sense conceptually to extend a baseUser to different types of users...

Reply:
yeah, it can be `baseUser`.. then it would be `superAdmin extends baseUser` impying that the superAdmin is extending the properties of any base user.

Reply:
Done ! PTAL